src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -69,29 +69,33 @@ protected function makeSureFieldHasEntity($field)
 
         // if there's a method on the model with this name
         if (method_exists($this->model, $field['name'])) {
-            $field['entity'] = $field['name'];
+            if ($this->model->checkIfMethodReturnRelation($field['name'])) {
+                $field['entity'] = $field['name'];
+            }
 
             return $field;
-        } // TODO: also check if that method is a relationship (returns Relation)
+        }
```

Comment:
Here's a thought. Do we actually _need_ to check that method is a relationship, if the developer passed it as the ```name``` of the field? Can't we just _assume_ it's a relationship?

Currently:
- ```name``` is the only mandatory attribute for fields;
- we require people to use the attribute name as the field ```name```

We could just:
- tell people that ```name``` should be either:
    - (A) the name of the attribute on the Model
    - or 
    - (B) the name of the relationship on the Model

So there are two ways to go with this, as far as I can tell:

## 1 - SAFEST - only determine stuff if the method return type is Relation

- if a method exists on the model, DON'T _assume_ it's a relationship

      - if the method has a return type, and the return type is Relation
             - get all info from there (```entity```, ```relation_type```, ```multiple```, ```model``` etc)
      - else
            - we can just set the ```entity```;


## 2 - MAY ALSO BE SAFE? - assume we can invoke/call the methods given as ```name```

- if a method exists on the model, DO _assume_ it's a relationship

      - if the method has a return type, and the return type is Relation
            - get all info from there (```entity```, ```relation_type```, ```multiple```, ```model``` etc)
      - else
            - fail;

---

By taking a look at ```CrudPanel::getRelationModel()``` I realise we've been doing something similar all along. 


---

What do you think @pxpm ? Is it not _safe_ to assume that if the developer has given the name of a method as the name of a field, we can use call it?

Reply:
I agree with you on this @tabacitu . Bare with me this needs to take into account the following scenarios from my understanding:

```php

$this->crud->addField('related_method');

//we can check if method exists in the current model, if it exists 
//I think it's safe to assume that it's a relation. 
//But if we want to be 100% sure, the safest would be solution 1, 
// but that will force the developer to setup return types in their relations.

$this->crud->addField('related_method.attribute');

$this->crud->addField('related_method.related_method'); //no attribute

$this->crud->addField('related_method.related_method.attribute'); //or any relation chain.
```

Also, we were forcing `$field['type'] = 'relationship'` if we found signs it beeing a relationship. 
I think that is something we should also avoid because we should probably add the ability to edit 1-1 relations on same crud defining fields like:

```php
$this->crud->addFields(
['name' => 'address.street],
['name' => 'address.city],
['name' => 'address.postal_code],
);

```

By default this fields should be text fields. But this only applies to some relation types, other types (hasMany, belongsToMany etc) the field should probably be a select. We can work that out in relationship.blade.php ?

Reply:
> Also, we were forcing $field['type'] = 'relationship' if we found signs it beeing a relationship. I think that is something we should also avoid because we should probably add the ability to edit 1-1 relations on same crud defining fields like:

Ugh... you're right, of course. Damn it. In this case, we can only force the "relationship" field if we know the very end of the name is a method. Right? Would this be a good indicator?

Reply:
I think at this point, where it 99% done, it would be a very good idea to create unit tests for all the use cases you mentioned above. So that we make sure they're all taken care of.